### PR TITLE
Some refactors and changes in structure

### PR DESCRIPTION
**Pull request by lpozo**
Hey Jim!

I hope you're doing great!

I've been playing with this tool and I've done some changes to the code. Would you consider a review and a potential merge? I can tell you in advance that you'll find a lot of changes. Check the commit messages and ask my whatever you'd like to know.

I also updated the tests.

I only have an issue with a regex that I couldn't fix because I'm not well-versed in regex. Left a TODO.

Thanks in advance,

Leodanis

### CODE REVIEW

**Comment by jima80525** [comment]
@lpozo  !  I'm happily surprised that anyone else is using this!  I've got a busy day, so I probably won't get to this PR today, but will definitely get to it in the next couple of days.  I almost certainly will take the changes as they are (might make some comments, but If I recall, this tool was hacked together without a lot of thought, so I doubt you could have made it worse :) 
Thanks!

**Comment by lpozo** [comment]
Great!

I also unified the `--jima` option with the bad words. You can revert this if you want :). We can provide default dictionaries of bad words and probably add an option to allow the user to add their own bad words and phrases using the CLI. This user-defined dictionary can live in a config file or something.

File: rplint/main.py [hunk 1]
```diff
@@ -1,86 +1,83 @@
...
 import re
 import string
+from pathlib import Path
 
-__version__ = "0.7.1"
+import click
 
+BAD_WORDS_DIR = Path(__file__).parent.parent / "dicts"
+TRUNCATE_LENGTH = 40
+RP_SYNTAX_HIGHLIGHTERS = ["cpp"]
```
Comment (jima80525) [inline rplint/main.py:10]: I don't think this constant is being used.

File: tasks.py [hunk 2]
```diff
@@ -1,10 +1,9 @@
...
-from invoke import task
+releasing rplint.
+"""
+
 import pytest
-import setuptools
-import sys
+from invoke import run, task
 
 VERSION = "0.7.1"
```
Comment (jima80525) [inline tasks.py]: If you run `invoke version` I *think* it will update these for you, which is a good idea.

**Review by jima80525 — COMMENTED** [review]
Super!  The only major request is to update the version number (`invoke version` should do it).  

Thanks for updating this - that's MUCH better than what I had written! 
:)

**Comment by lpozo** [comment]
Thanks for the review! I'll fix these issues and we could probably deploy the new version to PyPI.

**Comment by lpozo** [comment]
Hey Jim! I bumped the version. I made a few extra commits because I wasn't sure how the bump version option woked.

**Comment by lpozo** [comment]
I'd like to continue making changes. So, I'll probably open another PR soon.
- I'd like to move the command definition to `__main__.py`.
- I'd like to rename `main.py` to `checkers.py` or something similar. We'll put the checker classes in that file.
- Unify the bad words in a single file, probably a JSON file with keys for bad_words, bad_phrases and so on.
- I'd like to implement an option to add custom bad words and so on.
- I'll do a few renaming here and there too.

We can now update the PyPI package.

**Comment by jima80525** [comment]
hey!  Sorry I didn't get notified when you made changes.  I'll merge this now.  Thanks again for working on it!

**Pull request merged**